### ISSUE TITLE
Visual Prompt Tuning for Generative Transfer Learning

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#193]
# TL; DR

![image](https://user-images.githubusercontent.com/46308407/203983282-f7971ffa-4da1-4dc1-a02b-18e0d86a8eac.png)

- Train a parameter-efficient prompt-generating network that conditions on either class index or an instance to adapt a powerful pretrained generative ViT for an exotic domain.

# Motivation

- Transfer learning, despite being a key corner stone in ML that enables data-efficient adaptation, has been under-explored in generative setting.

# Formulation 

### Preliminary: AR and NAR Generative Transformer

![image](https://user-images.githubusercontent.com/46308407/203983333-77996835-8e95-4eaa-9399-ebb50a6c4bc7.png)

- Auto-regressive generative ViTs simply raster-scan patch tokens, from left to right and top to bottom.
- Non-autoregressive generative ViT synthesizes quantized image tokens from a group of [MASK] tokens (corresponding for the patch grid), but in a scheduled manner by only preserving patch tokens with high confidence.

### Parameter-efficient Prompt Generator

![image](https://user-images.githubusercontent.com/46308407/203983372-c5c7a33f-c4bb-4895-a6c2-71af41f5f986.png)

- Because producing adequate prompt tokens for a number of classes can be computationally burdensome, the authors propose a parameter-efficient design that *factorizes* the class/instance condition and token position, as well as a learnable factor to blend these two.

### Marquee Header

- Instance conditioning enables fine-grained control over synthesis as well as improves overall sample quality at the cost of diversity (of course).
- Class conditioning provides more general information, that can be potentially beneficial for diversity.
- Hence, the authors propose to interpolate between these two conditioning prompts (or two instance prompts) to strike a sweet balance between quality and diversity.

![image](https://user-images.githubusercontent.com/46308407/203984342-f32dcc10-25d4-4450-9f5e-e24d3663a410.png)

# Evaluation

![image](https://user-images.githubusercontent.com/46308407/203984376-d75656de-1e45-4420-a929-60d24b453980.png)
![image](https://user-images.githubusercontent.com/46308407/203984392-8b366a8e-4e52-453e-8afe-13fadbc1b86d.png)
![image](https://user-images.githubusercontent.com/46308407/203984400-c1920d58-e8d3-42a3-b7a5-09ac36ae7b49.png)
![image](https://user-images.githubusercontent.com/46308407/203984417-b4d9414c-7a59-4069-9a61-9b38c9444ab3.png)

- T-SNE visualization shows that instance prompt representations are grouped in accordance with the underlying class labels.
- This indicates that the prompt generator network has indeed learned the general semantics in the images.

![image](https://user-images.githubusercontent.com/46308407/203984589-07cb2043-8011-4a7e-9c78-5827d49166b0.png)
![image](https://user-images.githubusercontent.com/46308407/203984624-3bc5a562-d53b-4c17-a001-db1e07df3873.png)
![image](https://user-images.githubusercontent.com/46308407/203984648-f0af6f2e-56ad-4d95-a347-5a8be80fb014.png)
![image](https://user-images.githubusercontent.com/46308407/203984667-2a81eca2-e422-43af-b3df-c2367d7230ae.png)

# Inspiration

- With the advent of ViTs, popular techniques in NLP are overflowing into the vision domain. Prompt tuning is gaining great attention thanks to its compute efficiency and effectiveness.
- The idea of actively using instance conditioning was good. Were it not for that, this paper (report) might have been overly plain.
- This can be viewed as a hypernetwork framework in a sense (though not directly outputting the model parameters). Wonder it can be done more naively with a set of learnable parameters (prompts).
